nginx_module/src/http_request.rs

```diff
@@ -57,6 +60,59 @@ impl<'a, Ctx: Default> HttpRequestAndContext<'a, Ctx> {
             Ok((req, ctx))
         }
     }
+
+    // get raw pointer to request structure
+    pub fn get_request_pointer (
```

Comment:
A Rust reference (&mut self) is already supposed to not be null so there is no reason to check for null here, or to use a Result. The function that converts a raw pointer to a Rust reference should do this check, not this one.

Reply:
fixed